tools/zbdsniff

```diff
@@ -27,98 +52,113 @@ def getnetworkkey(packet):
+                if ord(decrypted[1]) < len(KEY_TYPE):
+                    print("    Key Type: " + KEY_TYPE[ord(decrypted[1])])
+                print("    Value: " + ":".join("{:02x}".format(ord(ch)) for ch in decrypted[2:18]))
 
-        # Check payload length, must be at least 35 bytes
-        # APS cmd | key type | key | sequence number | dest addr | src addr
-        if len(zapspayload) < 35:
-            return
+    except Exception, e:
+        if verbose:
+            print("[-] Error:" + str(e))
+        return
 
-        # Check for APS command identifier Transport Key (0x05)
-        if ord(zapspayload[0]) != 5:
-            return
 
-        # Transport Key Frame, get the key type.  Network Key is 0x01, no
-        # other keys should be sent in plaintext
-        if ord(zapspayload[1]) != 1:
-            print "Possible key or false positive?"
-            return
+# Define the command line options.
+parser = OptionParser()
+parser.add_option("-f", "--file", dest="filename",
+                  help="PCap file to process", metavar="FILE")
+parser.add_option("-d", "--dir", dest="directory",
+                  help="Directory of PCap files to process", metavar="DIR")
+parser.add_option("-k", "--transport-key", dest="transportKey",
+                  help="Transport Key for decryption")
+parser.add_option("-v", "--verbose",
+                  action="store_true", dest="verbose", default=False,
+                  help="Print detailed status messages to stdout")
 
-        # Reverse these fields
-        networkkey = zapspayload[2:18][::-1]
-        destaddr = zapspayload[19:27][::-1]
-        srcaddr = zapspayload[27:35][::-1]
-
-        print "NETWORK KEY FOUND: ",
-        for x in networkkey[0:15]:
-            sys.stdout.write("%02x:"%ord(x))
-        print "%02x" % ord(networkkey[15])
-        print "      (Wireshark): ",
-        for x in zapspayload[2:17]:
-            sys.stdout.write("%02x:"%ord(x))
-        print "%02x" % ord(zapspayload[17])
-
-        print "  Destination MAC Address: ",
-        for x in destaddr[0:7]:
-            sys.stdout.write("%02x:"%ord(x))
-        print "%02x" % ord(destaddr[7])
-
-        print "  Source MAC Address:      ",
-        for x in srcaddr[0:7]:
-            sys.stdout.write("%02x:"%ord(x))
-        print "%02x" % ord(srcaddr[7])
-
-    except Exception, e:
-        #print e
-        return
+(options, args) = parser.parse_args()
 
+if (not options.filename and not options.directory):
+    print("A packet capture file or directory must be specified")
+    sys.exit(1)
 
-if len(sys.argv) == 1:
-    usage()
-    sys.exit(0)
+if (not options.transportKey):
+    print("A transport key value must be specified")
+    sys.exit(1)
 
 files = []
-filecount = 0
+if options.filename:
+    files.append(options.filename)
 
-while len(sys.argv) > 1:
-    files.append(sys.argv.pop(1))
+if options.directory:
+    files += glob.glob(options.directory + os.sep + "*.pcap")
```

Comment:
prefer os.join() but this is the equivalent and is platform safe so will be OK

Reply:
Hmmm, I'll go for os.path.join() as that delegates the responsibility entirely.